test/TorchSharpTest/NN.cs

```diff
@@ -4714,6 +4714,12 @@ public void TestLayerNorm()
                     Assert.Equal(ones.shape, pooled.shape);
                 }
             }
+            if (torch.cuda.is_available()) {
+                using var pool = LayerNorm(new long[] { 12, 28, 28 }, device: torch.CPU);
+
+                using var pool1 = pool.to(torch.CUDA);
+                Assert.Equal(DeviceType.CUDA, pool1.weight.device_type);
+            }
```

Comment:
I don't think this unit test covers the bug reported. 
The issue was the LayerNorm not being recognized as a submodule, so perhaps placing it inside a sequential?

Reply:
But that wasn't the actual issue. 

The issue was that a) it used IntPtr.Zero in the base constructor call, and then (once that was addressed) b) that the 'weight' and 'bias' properties weren't set, so the parameters fetched by the properties were the old ones. The ComponentName attribute fixed that.

Reply:
Got it. So maybe just add a check to clarify that it isn't invalid?

```cs
Assert.False(pool.IsInvalid)
```

Reply:
`Module.handle` is not public, so no can do. However, any call to `to()` will blow up if it's invalid, so the fact that the call does not should prove the same point.

Reply:
Good point, I missed that part